session.go

```diff
@@ -66,8 +67,8 @@ type Session interface {
 	AffectedRows() uint64                        // Affected rows by latest executed stmt.
 	Execute(sql string) ([]ast.RecordSet, error) // Execute a sql statement.
 	String() string                              // String is used to debug.
-	CommitTxn() error
-	RollbackTxn() error
+	CommitTxn(goctx.Context) error
```

Comment:
Can we not adding `goctx.Context`, always use `Session.GoCtx` in implementation?

Reply:
It's better to add this argument, and avoid store ctx in the object if possible.
For example, session start a txn and store the txn to `session.GoCtx`,
then call function `F(ctx)`,
`F(ctx)` call `CommitTxn()... span parent of `CommitTxn()` should be `F(ctx)`, not the `session.GoCtx`